src/bundle/DependencyInjection/Configuration.php

```diff
@@ -48,6 +48,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->scalarNode('ip_whitelist_provider')->defaultValue('scheb_two_factor.default_ip_whitelist_provider')->end()
                 ->scalarNode('two_factor_token_factory')->defaultValue('scheb_two_factor.default_token_factory')->end()
                 ->scalarNode('two_factor_condition')->defaultNull()->end()
+                ->scalarNode('rate_limiter')->defaultNull()->end()
```

Comment:
I was rather thinking about a true/false switch here. I wouldn't know what to put here ;)

Reply:
I intended for a rate limiter service to be put here, since those are highly configurable and I don't want to decide for the user what kind of rate limiter to use.

Reply:
Could you give me an example how to use this? How would such a RateLimiterFactory service be configured?

Could be that I didn't fully get it and my comments don't make much sense 🙈

Reply:
Say for example you have this in your `rate_limiter.yaml`:

```
framework:
    rate_limiter:
        mfa:
            policy: 'token_bucket'
            limit: 4
            rate:
                interval: '5 minutes'
                amount: 1
            cache_pool: 'some.cache.pool'
            lock_factory: 'some.lock.factory'
```

Then in `scheb_2fa.yaml` you can have something like:
```
scheb_two_factor:
    rate_limiter: 'limiter.mfa'
```

Now you will have your MFA attempts rate limited via the 'token bucket' strategy, with max 4 attempts and 1 new attempt every 5 minutes. The component documentation does a much better job at explaining how it works and how the different limiting strategies work: https://symfony.com/doc/current/rate_limiter.html.

Reply:
Ahh, I see. I wasn't ware you have all these configurations options on the rate limiter. Makes much more sense now!

Under these conditions we can keep everything as-is. _Though_, we need to have good defaults for the rate limiter in the documentation, as most people would just go with it.